### ISSUE TITLE
VAE training using test loss to decay beta

### ISSUE THREAD

**Issue by Ajoo** [amazon-science/tabsyn#31]
Hi,

I noticed that while training the VAE for TabSyn, a loss computed on the test set seems to be used to decay the learning rate and $\beta$. Is this intentional?

The issue is in the following snippet:

https://github.com/amazon-science/tabsyn/blob/cb5ac0f74ec36ee88e7a974a393dfbef50d42da7/tabsyn/vae/main.py#L165-L187

The loss used to decay `lr` and `beta` is computed in lines 168 and 169 on the test set.
Furthermore, this loss includes only the cross-entropy component for the categoricals as the mse loss is zeroed out in line 169. This seems like an odd choice given that in some datasets there might be very few categoricals.

**Comment by TanguyUrvoy** [comment]
I do agree with these remarks:
- using the test-set to calibrate beta and learning rate is a data-leak that may inflate artificially the model's performance. Proper methodology is to use a validation set. I hope this was not done intentionally.
- The use of cross-entropy only as a validation loss (renamed "train_loss" ??) is also very strange and not commented in the paper.

I may add a few other remarks on this codebase:
- The train_val_test_split function in process_dataset.py is also a data-leak: it bypasses the missing categories in test set by selecting only splits where all categories are covered by train set.
- There is a lot of dead code and redundancy that could be removed safely
